src/SprykerEco/Zed/AfterPay/Business/Payment/Filter/TwoStepAuthorizePaymentMethodsFilter.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\AfterPay\Business\Payment\Filter;
+
+use ArrayObject;
+use Generated\Shared\Transfer\AfterPayAvailablePaymentMethodsTransfer;
+use Generated\Shared\Transfer\PaymentMethodsTransfer;
+use Generated\Shared\Transfer\PaymentMethodTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use SprykerEco\Zed\AfterPay\AfterPayConfig;
+use SprykerEco\Shared\AfterPay\AfterPayConfig as SharedAfterPayConfig;
+use SprykerEco\Zed\AfterPay\Business\Payment\Filter\Provider\AfterPayPaymentMethodsProviderInterface;
+
+class TwoStepAuthorizePaymentMethodsFilter implements AfterPayPaymentMethodsFilterInterface
+{
+    /**
+     * @param \Generated\Shared\Transfer\PaymentMethodsTransfer $paymentMethodsTransfer
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\PaymentMethodsTransfer
+     */
+    public function filterPaymentMethods(
+        PaymentMethodsTransfer $paymentMethodsTransfer,
+        QuoteTransfer $quoteTransfer
+    ): PaymentMethodsTransfer {
+        $result = new ArrayObject();
+
+        foreach ($paymentMethodsTransfer->getMethods() as $paymentMethodTransfer) {
+            if($this->isPaymentProviderAfterPay($paymentMethodTransfer) &&
```

Comment:
Please move conditions to separate method, they have to be clear and easy to understand.